src/ripple/app/tests/PayChan_test.cpp

```diff
@@ -0,0 +1,689 @@
+            auto const bob = Account ("bob");
+            env.fund (XRP (10000), alice, bob);
+            auto const pk = alice.pk ();
+            auto const settleDelay = 3600s;
+            auto const channelFunds = XRP (1000);
+            env (create (alice, bob, channelFunds, settleDelay, pk));
+            auto const chan = channel (*env.current (), alice, bob);
+            expect (channelExists (*env.current (), chan));
+
+            env (fset (bob, asfDisallowXRP));
+            auto const preBob = env.balance (bob);
+            auto const reqBal = XRP (500).value();
+            env (claim (alice, chan, reqBal, reqBal), ter(tecNO_TARGET));
+        }
+    }
+
+    void
+    testDstTag ()
+    {
+        // auth amount defaults to balance if not present
+        testcase ("Dst Tag");
+        using namespace jtx;
+        using namespace std::literals::chrono_literals;
+        // Create a channel where dst disallows XRP
+        Env env (*this, features (featurePayChan));
+        auto const alice = Account ("alice");
+        auto const bob = Account ("bob");
+        env.fund (XRP (10000), alice, bob);
+        env (fset (bob, asfRequireDest));
+        auto const pk = alice.pk ();
+        auto const settleDelay = 3600s;
+        auto const channelFunds = XRP (1000);
+        env (create (alice, bob, channelFunds, settleDelay, pk),
+            ter (tecDST_TAG_NEEDED));
+        expect (!channelExists (
+            *env.current (), channel (*env.current (), alice, bob)));
+        env (
+            create (alice, bob, channelFunds, settleDelay, pk, boost::none, 1));
+        expect (channelExists (
+            *env.current (), channel (*env.current (), alice, bob)));
+    }
+
+    void
+    testMultiple ()
+    {
+        // auth amount defaults to balance if not present
+        testcase ("Multiple channels to the same account");
+        using namespace jtx;
+        using namespace std::literals::chrono_literals;
+        Env env (*this, features (featurePayChan));
+        auto const alice = Account ("alice");
+        auto const bob = Account ("bob");
+        env.fund (XRP (10000), alice, bob);
+        auto const pk = alice.pk ();
+        auto const settleDelay = 3600s;
+        auto const channelFunds = XRP (1000);
+        env (create (alice, bob, channelFunds, settleDelay, pk));
+        auto const chan1 = channel (*env.current (), alice, bob);
+        expect (channelExists (*env.current (), chan1));
+        env (create (alice, bob, channelFunds, settleDelay, pk));
+        auto const chan2 = channel (*env.current (), alice, bob);
+        expect (channelExists (*env.current (), chan2));
+        expect (chan1 != chan2);
+    }
+
+
+    void
+    run () override
+    {
+        testSimple ();
+        testCancelAfter ();
+        testSettleDelay ();
+        testExpiration ();
+        testCloseDry ();
+        testDefaultAmount ();
+        testDisallowXRP ();
+        testDstTag ();
```

Comment:
There are situations where the dst tag is required, and we test for those. There are no analogous tests for src tag.